packages/core/src/node/messaging/connection.ts

```diff
@@ -40,10 +44,37 @@ export interface OnOpen {
 }
 
 export function openSocket(options: IServerOptions, onOpen: OnOpen): void {
+
     const wss = new ws.Server({
         noServer: true,
         perMessageDeflate: false
     });
+
+    wss.on('connection', (ws: ws) => {
+
+        const extWs = ws as ExtWebSocket;
+        extWs.isAlive = true;
+
+        extWs.on('pong', () => {
+            extWs.isAlive = true;
+        });
+
+    });
+
+    setInterval((ws: ExtWebSocket) => {
+
+        const extWs = ws as ExtWebSocket;
+
+        if (extWs.isAlive === false) {
+            ws.terminate();
+            return;
+        }
+
+        extWs.isAlive = false;
+        ws.ping(null, undefined);
+
+    }, 30000);
+
```

Comment:
As mentioned above, wss does not emit "connection" when the web socket is created and opened.  So the listener above will never trigger. A way around that is to emit it ourselves. To do so we have to change the code below a bit. Something like:

```typescript
if (webSocket.readyState === webSocket.OPEN) {
   onOpen(webSocket, request, socket, head);
   wss.emit('connection', webSocket);
} else {
   webSocket.on('open', () => {
      onOpen(webSocket, request, socket, head);
      wss.emit('connection', webSocket);
   });
}
```

Reply:
excellent I kind of felt like it needed something in that area but I was not sure what, I will get this in and tested.